[rocprofiler-compute] Update Documentation for Strix Halo (#4794)
ROCm/rocm-systems: projects/rocprofiler-compute/docs/conceptual/rdna/tcp-cache.rst

.. meta::

>    :description: ROCm Compute Profiler RDNA3.5 TCP (GL0 / vector L0) metrics

Reviewer: Title incorrect, or not worded well. "vector L0" I believe should be "vector L1 data cache", but this might be the old reference, renamed to GL0 in gfx10+.  I think we have poor naming in general where "L0 global cache" could easily be confused with the per-CU L0 cache (which is related to the SA-level cache)

   :keywords: ROCm Compute Profiler, RDNA, gfx1151, TCP, GL0, L0

.. _rdna-tcp:

=========
TCP (GL0)
=========

> On gfx1151, TCP is the vector L0 cache (RDNA GL0) in front of GL1C. For GL1C (L1)

Reviewer: ```suggestion
On gfx1151, TCP is the vector L1 data cache (RDNA GL0) in front of GL1C. For GL1C (L1)
```

TCP cache (panel)
=================

TCP Utilization
---------------

.. tab-set::

   .. tab-item:: RDNA 3.5 (gfx1151)
      :selected:

      .. jinja:: rdna1151-tcp-utilization-gfx1151
         :file: _templates/metrics_table.j2

TCP Request Statistics
----------------------

.. tab-set::

   .. tab-item:: RDNA 3.5 (gfx1151)
      :selected:

      .. jinja:: rdna1151-tcp-request-statistics-gfx1151
         :file: _templates/metrics_table.j2

TCP Cache Performance
---------------------

.. tab-set::

   .. tab-item:: RDNA 3.5 (gfx1151)
      :selected:

      .. jinja:: rdna1151-tcp-cache-performance-gfx1151
         :file: _templates/metrics_table.j2

TCP-GL1 interface
-----------------

.. tab-set::

   .. tab-item:: RDNA 3.5 (gfx1151)
      :selected:

      .. jinja:: rdna1151-tcp-tcp-gl1-interface-gfx1151
         :file: _templates/metrics_table.j2

TCP Stalls
----------

.. tab-set::

   .. tab-item:: RDNA 3.5 (gfx1151)
      :selected:

      .. jinja:: rdna1151-tcp-stalls-gfx1151
         :file: _templates/metrics_table.j2

Memory Chart: path up to GL1
============================

The following Memory Chart tables align with the on-screen flow through instruction

> and scalar paths, TCP (vector L0), LDS, and the TCP-GL1 interface.

Reviewer: Same vector L0 issue.

```suggestion
and scalar paths, TCP (GL0), LDS, and the TCP-GL1 interface.
```

Memory chart - Instruction Cache
---------------------------------

.. tab-set::

   .. tab-item:: RDNA 3.5 (gfx1151)
      :selected:

      .. jinja:: rdna1151-memory-chart-instruction-cache-gfx1151
         :file: _templates/metrics_table.j2

Memory chart - Scalar Data Cache
---------------------------------

.. tab-set::

   .. tab-item:: RDNA 3.5 (gfx1151)
      :selected:

      .. jinja:: rdna1151-memory-chart-scalar-data-cache-gfx1151
         :file: _templates/metrics_table.j2

> Memory chart - TCP Cache (Vector L0)

Reviewer: ```suggestion
Memory chart - TCP Cache (Vector Data Cache)
```

.. meta::
   :description: ROCm Compute Profiler RDNA3.5 TCP (GL0) metrics
   :keywords: ROCm Compute Profiler, RDNA, gfx1151, TCP, GL0, L0

.. _rdna-tcp:

===
GL0
===

On gfx1151, TCP is the vector L1 data cache (RDNA GL0) in front of GL1C. For GL1C
panels and the GL1C Memory Chart table, see :doc:`gl1-cache`. The handoff toward
GL2C is under :doc:`gl2-cache`.

**GL0 is the same as TCP on current architecture. We keep using the name "TCP" on this
page for consistency with the counters' names.**

TCP cache panels
================

> TCP Utilization

Reviewer 2: ```suggestion
TCP utilization
```

Reviewer 3: resolved in 206673250825b2c00edf3c245b5c52b2737ff869

---------------

.. tab-set::

   .. tab-item:: RDNA 3.5 (gfx1151)
      :selected:

      .. jinja:: rdna1151-tcp-utilization-gfx1151
         :file: _templates/metrics_table.j2

> TCP Request Statistics

Reviewer 2: ```suggestion
TCP request statistics
```

Reviewer 3: resolved in 206673250825b2c00edf3c245b5c52b2737ff869

----------------------

.. tab-set::

   .. tab-item:: RDNA 3.5 (gfx1151)
      :selected:

      .. jinja:: rdna1151-tcp-request-statistics-gfx1151
         :file: _templates/metrics_table.j2

> TCP Cache Performance

Reviewer 2: ```suggestion
TCP cache performance
```

Reviewer 3: resolved in 206673250825b2c00edf3c245b5c52b2737ff869

---------------------

.. tab-set::

   .. tab-item:: RDNA 3.5 (gfx1151)
      :selected:

      .. jinja:: rdna1151-tcp-cache-performance-gfx1151
         :file: _templates/metrics_table.j2

TCP-GL1 interface
-----------------

.. tab-set::

   .. tab-item:: RDNA 3.5 (gfx1151)
      :selected:

      .. jinja:: rdna1151-tcp-tcp-gl1-interface-gfx1151
         :file: _templates/metrics_table.j2

> TCP Stalls

Reviewer 2: ```suggestion
TCP stalls
```

Reviewer 3: resolved in 206673250825b2c00edf3c245b5c52b2737ff869

----------

.. tab-set::

   .. tab-item:: RDNA 3.5 (gfx1151)
      :selected:

      .. jinja:: rdna1151-tcp-stalls-gfx1151
         :file: _templates/metrics_table.j2

> Memory Chart: path up to GL1

Reviewer 2: ```suggestion
Memory chart: path up to GL1
```

Reviewer 3: resolved in 206673250825b2c00edf3c245b5c52b2737ff869

============================

The following Memory Chart tables align with the on-screen flow through instruction
and scalar paths, TCP (GL0), LDS, and the TCP-GL1 interface.

> Memory chart - Instruction Cache

Reviewer 2: ```suggestion
Memory chart - instruction cache
```

Reviewer 3: resolved in 206673250825b2c00edf3c245b5c52b2737ff869

---------------------------------

.. tab-set::

   .. tab-item:: RDNA 3.5 (gfx1151)
      :selected:

      .. jinja:: rdna1151-memory-chart-instruction-cache-gfx1151
         :file: _templates/metrics_table.j2

> Memory chart - Scalar Data Cache

Reviewer 2: ```suggestion
Memory chart - scalar data cache
```

Reviewer 3: resolved in 206673250825b2c00edf3c245b5c52b2737ff869

---------------------------------

.. tab-set::

   .. tab-item:: RDNA 3.5 (gfx1151)
      :selected:

      .. jinja:: rdna1151-memory-chart-scalar-data-cache-gfx1151
         :file: _templates/metrics_table.j2

> Memory chart - TCP Cache (Vector Data Cache)

Reviewer 2: ```suggestion
Memory chart - TCP cache (vector data cache)
```

Reviewer 3: resolved in 206673250825b2c00edf3c245b5c52b2737ff869

---------------------------------------------

.. tab-set::

   .. tab-item:: RDNA 3.5 (gfx1151)
      :selected:

      .. jinja:: rdna1151-memory-chart-tcp-cache-vector-l0-gfx1151
         :file: _templates/metrics_table.j2

> Memory chart - LDS (Local Data Share)

Reviewer 2: ```suggestion
Memory chart - LDS (local data share)
```

Reviewer 3: resolved in 206673250825b2c00edf3c245b5c52b2737ff869

--------------------------------------

.. tab-set::

   .. tab-item:: RDNA 3.5 (gfx1151)
      :selected:

      .. jinja:: rdna1151-memory-chart-lds-local-data-share-gfx1151
         :file: _templates/metrics_table.j2

> Memory chart - TCP-GL1 Interface

Reviewer 2: ```suggestion
Memory chart - TCP-GL1 interface
```

Reviewer 3: resolved in 206673250825b2c00edf3c245b5c52b2737ff869
